openstack/standalone-cinder/pkg/volumeservice/actions.go

```diff
@@ -109,6 +112,22 @@ func ConnectCinderVolume(vs *gophercloud.ServiceClient, volumeID string) (Volume
 	return rcv.ConnectionInfo, nil
 }
 
+// AttachCinderVolume marks the volume as attached in the cinder database.
+func AttachCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error {
+	opts := volumeactions.AttachOpts{
+		MountPoint: attachMountPoint,
+		HostName:   attachHostName,
+		Mode:       volumeactions.ReadWrite,
```

Comment:
not sure if mode is needed, cannot find it in cinder api https://developer.openstack.org/api-ref/block-storage/v3/#attach-volume-to-a-server

Reply:
From my testing it seemed to be required.